OSDOCS#10734: AWS Public IPv4 cost mitigation (#76834)
openshift/openshift-docs: modules/installation-configuration-parameters.adoc

For clusters that use AWS Local Zones, you must add AWS Local Zone subnets to this list to ensure edge machine pool creation.
|Valid subnet IDs.

|platform:
  aws:
    publicIpv4Pool:

> |IPv4 pool name from the public IPv4 pools in the AWS account to configure the cluster with. Your AWS account must be provisioned for public IPv4 pool. You must have 3n+1 pools available where n is the total number of AWS zones used to deploy the cluster.

Reviewer: ```suggestion
|Public IPv4 Pool ID used to allocate Elastic IPs when `publish` is set to `External`. The pool must be provisioned and advertised in the same AWS account and region of the cluster. You must have (2*n + 1) IPv4 available in the pool where `n` is the total number of AWS zones used to deploy the Network Load Balancer for API, NAT Gateways, and Bootstrap node. For more information about AWS BYO IPv4, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html#byoip-onboard[Onboard your BYOIP]
```

I suggest to keep the AWS doc reference to get users aware of AWS BYOIPv4 feature (it is a common mistake when talking about BYO IP because it can be IPAM, CoIP, etc).

Author: and not 3n+1 @mtulio can you confirm that it's 2n+1 as started [here](https://github.com/openshift/installer/blob/master/docs/user/aws/custom_public_ipv4.md)?

> |The public IPv4 pool ID that is used to allocate Elastic IPs (EIPs) when `publish` is set to `External`. You must provision and advertise the pool in the same AWS account and region of the cluster. You must have 2n + 1 IPv4 available in the pool where n is the total number of AWS zones used to deploy the Network Load Balancer (NLB) for API, NAT gateways, and bootstrap node. For more information about Bring your own IP addresses (BYOIP) in AWS, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html#byoip-onboard[Onboard your BYOIP]

Reviewer: ```suggestion
|The public IPv4 pool ID that is used to allocate Elastic IPs (EIPs) when `publish` is set to `External`. You must provision and advertise the pool in the same AWS account and region of the cluster. You must have 2n + 1 IPv4 available in the pool where n is the total number of AWS zones used to deploy the Network Load Balancer (NLB) for API, NAT gateways, and bootstrap node. For more information about Bring your own Public IP addresses (BYO Public IPv4) in AWS, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html#byoip-onboard[Onboard your BYOIP]
```

@shubham-pampattiwar we are also discussion [here][slack] to add additional permissions here: https://docs.openshift.com/container-platform/4.15/installing/installing_aws/installing-aws-account.html#installation-aws-permissions_installing-aws-account

A new section must be added when the feature is used, it is checked in [permissions.go]:
- https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/aws/permissions.go#L47C2-L47C26
- https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/aws/permissions.go#L269-L274

@r4f4 any additional thoughts here?

cc @bscott-rh

Reviewer 2: I think that's it: mentioning in the docs that 2 new permissions are required when enabling the feature.

For clusters that use AWS Local Zones, you must add AWS Local Zone subnets to this list to ensure edge machine pool creation.
|Valid subnet IDs.

|platform:
  aws:
    publicIpv4Pool:

> |The public IPv4 pool ID that is used to allocate Elastic IPs (EIPs) when `publish` is set to `External`. You must provision and advertise the pool in the same AWS account and region of the cluster. You must ensure that you have 2n + 1 IPv4 available in the pool where n is the total number of AWS zones used to deploy the Network Load Balancer (NLB) for API, NAT gateways, and bootstrap node. For more information about bring your own IP addresses (BYOIP) in AWS, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html#byoip-onboard[Onboard your BYOIP].

Reviewer 3: ```suggestion
|The public IPv4 pool ID that is used to allocate Elastic IPs (EIPs) when `publish` is set to `External`. You must provision and advertise the pool in the same {aws-full} account and region of the cluster. You must ensure that you have 2n + 1 IPv4 available in the pool where _n_ is the total number of {aws-short} zones used to deploy the Network Load Balancer (NLB) for API, NAT gateways, and a bootstrap node. For more information about bring your own IP addresses (BYOIP) in {aws-short}, see link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html#byoip-onboard[Onboard your BYOIP] ({aws-short} documentation).
```

| A valid link:https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-public-ipv4-pools.html[public IPv4 pool id]

> [NOTE]

Reviewer 3: Ideally parameterize AWS as attributes exist: 
![Screenshot from 2024-06-19 16-23-30](https://github.com/openshift/openshift-docs/assets/57954076/f215716b-9e67-4ebf-b49f-d4937e61b8ed)

Author: I see. Looks like the entire assembly doesn't use attributes for AWS. Thanks!
